scheduler/spread.go

```diff
@@ -193,20 +182,31 @@ func evenSpreadScoreBoost(pset *propertySet, option *structs.Node) float64 {
 			maxCount = value
 		}
 	}
+
+	// calculate boost based on delta between the current and the minimum
+	deltaBoost := 1.0
+	if minCount == 0 {
+		deltaBoost = -1.0
+	} else {
+		delta := int(minCount - currentAttributeCount)
+		deltaBoost = float64(delta) / float64(minCount)
+	}
 	if currentAttributeCount < minCount {
-		// Small positive boost for attributes with min count
-		return evenSpreadBoost
+		// positive boost for attributes with min count
+		return deltaBoost
 	} else if currentAttributeCount > minCount {
 		// Negative boost if attribute count is greater than minimum
-		// This is so that other nodes will get a preference over this one
-		return evenSpreadPenalty
+		return deltaBoost
 	} else {
-		// When min and max are same the current distribution is even
-		// so we penalize
+		// When min and current value are the same we return the maximum
+		// possible boost or penalty
 		if minCount == maxCount {
-			return evenSpreadPenalty
+			// Maximum possible penalty when the distribution is even
+			return -1.0
 		} else {
-			return evenSpreadBoost
+			// Maximum possible boost when there is another attribute with
+			// more allocations
+			return 1.0
```

Comment:
Why isn't this delta based from the max? You could imagine attributes with the following alloc counts:

{a1: 2, a2: 3, a3: 4}
min = 2, max = 4.

a1 should have. a score higher than a2?